crates/iceberg/src/spec/values.rs

```diff
@@ -105,6 +108,115 @@ pub struct Datum {
     literal: PrimitiveLiteral,
 }
 
+impl Serialize for Datum {
+    fn serialize<S: serde::Serializer>(
+        &self,
+        serializer: S,
+    ) -> std::result::Result<S::Ok, S::Error> {
+        let mut strucet_ser = serializer
+            .serialize_struct("Datum", 2)
+            .map_err(serde::ser::Error::custom)?;
+        strucet_ser
+            .serialize_field("type", &self.r#type)
+            .map_err(serde::ser::Error::custom)?;
+        strucet_ser
+            .serialize_field(
+                "literal",
+                &RawLiteral::try_from(
```

Comment:
But we need the type info to convert to/from RawLiteral. For serialize in PrimitiveLiteral, we don't need the type info. But for deserialize, even in PrimitiveLiteral, we need the type info. e.g. https://github.com/apache/iceberg-rust/blob/7f286b56031b1ed9a9fb7d18aeb83abd736e44c4/crates/iceberg/src/spec/values.rs#L2440. That's why we can't implement Ser/De for Literal directly.

Reply:
One thing may not related this PR.

https://github.com/apache/iceberg-rust/blob/806840729c1dd886624f6ce3cb132d0912befb4b/crates/iceberg/src/spec/manifest.rs#L954

I find that for de/serialize, we need the type info. Maybe using Struct with StructType here is more appropriate. (For now, Datum doesn't support Struct). So that we can convert it between RawLiteral directly.